### PR TITLE
Also deploy to the new server

### PR DESCRIPTION
**Pull request by foolip**
Keep the old until the transition is complete. commit-snapshots and
anything else missing will be copied over manually.

### CODE REVIEW

**Comment by annevk** [comment]
If we're going to use rsync shouldn't we use the variant that removes files but excludes the commit-snapshots directory? And maybe the branch-snapshots directory that's still in use sometimes?

**Comment by annevk** [comment]
(It might be a good idea to keep branch-snapshots alive only for the duration of the PR, but that should be discussed separately I think.)

**Comment by foolip** [comment]
> If we're going to use rsync shouldn't we use the variant that removes files but excludes the commit-snapshots directory?

Is that possible in a single command? All that I could come up with is first excluding commit-snapshots and branch-snapshots from sync and deletion, and then transferring them in a second command. Would you like me to do that? What makes me nervous is that adding a small typo to it after the fact could blow away all of the data.

**Comment by annevk** [comment]
Yeah, it's what HTML uses: https://github.com/whatwg/html-build/blob/master/ci-deploy/inside-container.sh (first rsync command in that resource)

**Comment by foolip** [comment]
OK, I'll go with that.

File: build/deploy.sh [hunk 1]
```diff
@@ -118,4 +120,15 @@ if [[ "$TRAVIS" == "true" ]]; then
...
     scp -r -o UserKnownHostsFile=known_hosts "$WEB_ROOT" "$DEPLOY_USER@$SERVER":
+
+    # rsync to the (new) WHATWG server, without deleting anything
+    echo "$NEW_SERVER $NEW_SERVER_PUBLIC_KEY" > known_hosts
+    rsync --rsh="ssh -o UserKnownHostsFile=known_hosts" --verbose \
+          --archive --chmod=D755,F644 --compress \
+          "$WEB_ROOT" deploy@$NEW_SERVER:/var/www/
+    # now delete extraneous files, excluding $COMMITS_DIR and $BRANCHES_DIR
+    rsync --rsh="ssh -o UserKnownHostsFile=known_hosts" --verbose \
+          --existing --ignore-existing --delete \
```
Comment (foolip) [inline build/deploy.sh]: This probably looks odd. From the man page: "If this option is combined with the --ignore-existing option, no files will be updated (which can be useful if all you want to do is delete extraneous files)."

Comment (annevk) [reply build/deploy.sh]: Why would we not want to update files?

Comment (foolip) [reply build/deploy.sh]: The first command does all of the updating, the second only removes.

**Comment by foolip** [comment]
No spec has changed since this landed, so to see if it worked I restarted the build of dom on the master branch:
https://travis-ci.org/whatwg/dom/jobs/268019015

That failed with "rsync: Invalid argument passed to --chmod (D755,F644)"

That same thing works in https://github.com/whatwg/books/blob/master/deploy.sh, and AFAICT the difference is between Travis's trusty and precise environments. We haven't required any particular environment, so I don't understand why dom is getting trusty and some others precise. Trusty is becoming the default, so as a workaround I suggest we just ask for it in .travis.yml where this problem pops up.

**Pull request merged**